scripts/updateRiskOracleConfig.ts

```diff
@@ -1,51 +1,85 @@
-    // },
-    {
+  if (!riskOracleConfig) {
+    throw new Error("No configuration found for the current network");
+  }
+
+  const syncConfigKeys = [];
+  
+  for (const marketAddress in riskOracleConfig.markets) {
+    const marketConfig = riskOracleConfig.markets[marketAddress];
+    syncConfigKeys.push({
+      baseKey: keys.SYNC_CONFIG_MARKET_DISABLED,
+      data: encodeData(["address"], [marketAddress]),
+      newValue: marketConfig.syncConfigMarketDisabled,
+      description: `market: ${marketAddress}, market disabled: ${marketConfig.syncConfigMarketDisabled}`,
+    });
+    for (const marketParameter in marketConfig.marketParameters) {
+      syncConfigKeys.push({
+        baseKey: keys.SYNC_CONFIG_MARKET_PARAMETER_DISABLED,
+        data: encodeData(["address", "string"], [marketAddress, marketParameter]),
+        newValue: marketConfig.marketParameters[marketParameter],
+        description: `market: ${marketAddress}, parameter: ${marketParameter}, parameter disabled for market: ${marketConfig.marketParameters[marketParameter]}`,
+      });
+    }
+  }
+
+  for (const parameter in riskOracleConfig.parameters) {
+    syncConfigKeys.push({
       baseKey: keys.SYNC_CONFIG_PARAMETER_DISABLED,
-      data: encodeData(["string"], ["maxLongTokenPoolAmount"]),
-    },
-    // {
-    //   baseKey: keys.SYNC_CONFIG_MARKET_PARAMETER_DISABLED,
-    //   data: encodeData(["address", "string"], ["0x352f684ab9e97a6321a13CF03A61316B681D9fD2", "maxLongTokenPoolAmount"]),
-    // },
-  ];
-
-  const isDisabled = process.env.IS_DISABLED;
-  if (isDisabled === undefined) {
-    throw new Error("IS_DISABLED not specified");
+      data: encodeData(["string"], [parameter]),
+      newValue: riskOracleConfig.parameters[parameter],
+      description: `parameter: ${parameter}, parameter disabled: ${riskOracleConfig.parameters[parameter]}`,
+    });
   }
 
   const multicallWriteParams = [];
 
   for (const syncConfigKey of syncConfigKeys) {
-    multicallWriteParams.push(
-      config.interface.encodeFunctionData("setBool", [
-        syncConfigKey.baseKey,
-        syncConfigKey.data,
-        isDisabled === "true" ? true : false,
-      ])
-    );
+    const baseKey = syncConfigKey.baseKey;
+    const data = syncConfigKey.data;
+    const key = getFullKey(baseKey, data);
+
+    const newValue = syncConfigKey.newValue;
+    const oldValue = await dataStore.getBool(key);
+
+    if (newValue !== oldValue) {
+      multicallWriteParams.push(
+        config.interface.encodeFunctionData("setBool", [
+          baseKey,
+          data,
+          newValue,
+        ])
+      );
+      console.info(`updating: ${syncConfigKey.description}`);
+    }
   }
 
-  console.info(`updating ${multicallWriteParams.length} features`);
+  console.info(`updating for ${multicallWriteParams.length} keys`);
   console.info("multicallWriteParams", multicallWriteParams);
 
-  if (process.env.WRITE) {
+  if (multicallWriteParams.length > 0 && process.env.WRITE === "true") {
```

Comment:
can it be similar to updateMarketConfigUtils?

https://github.com/gmx-io/gmx-synthetics/blob/updates/scripts/updateMarketConfigUtils.ts#L555

there is only one check `if (write) ...`

Reply:
Sure, the additional checks were mainly helpful for debugging but I will remove them and better align with the provided example